common/types/hashes.go

```diff
@@ -227,8 +221,21 @@ func (h Hash32) ToHash20() (h20 Hash20) {
 	return
 }
 
+type hexBytesStringer []byte
+
+func (l hexBytesStringer) String() string {
+	return hex.EncodeToString(l)
+}
+
 // Field returns a log field. Implements the LoggableField interface.
-func (h Hash32) Field() log.Field { return log.String("hash", hex.EncodeToString(h[:])) }
+func (h Hash20) Field() log.Field {
+	return log.Stringer("hash", hexBytesStringer(h[:]))
```

Comment:
Why not?

```suggestion
	return log.Stringer("hash", h)
```

Reply:
I know. I didn't want to change the format in the logs. AFAIR, nowhere in logs do we use the "0x" prefix. Maybe it would be fine to change the `String()` implementation to not add the prefix?

Reply:
I don't think it matters much for logs, but we could check if `String()` is actually used anywhere already and if not change it.